src/mpyl/steps/deploy/k8s/__init__.py

```diff
@@ -57,7 +105,30 @@ def upsert_namespace(
         logger.info(f"Found namespace {namespace}")
 
 
-def deploy_helm_chart(
+def render_manifests(chart: dict[str, CustomResourceDefinition]):
+    result = f"{GENERATED_WARNING}\n"
+    for name, template_content in chart.items():
+        manifest = render_crd(name, template_content)
+        result += manifest
+    return result
+
+
+def render_crd(name: str, crd: CustomResourceDefinition):
+    return f"---\n# {name}\n{to_yaml(crd)}"
+
+
+def write_manifest(
+    release_name: str, target_path: Path, chart: dict[str, CustomResourceDefinition]
+) -> Path:
+    if not target_path.exists():
+        os.makedirs(target_path, exist_ok=True)
+    manifests = render_manifests(chart)
+    manifest_file = target_path / f"{release_name}-manifest.yml"
+    manifest_file.write_text(manifests, "utf-8")
+    return manifest_file
+
+
+def deploy_helm_chart(  # pylint: disable=too-many-locals
```

Comment:
This function now does 3 different things, should we maybe split it up?

Reply:
I see what you mean. I wouldn't want to increase the burden on the calling side though. It's sort of a strategy pattern without the polymorphic class. We could use that approach, but perhaps a bit over the top.
I've extracted some methods at least